xbmc/rendering/dx/DeviceResources.cpp

```diff
@@ -1370,13 +1374,28 @@ HDR_STATUS DX::DeviceResources::ToggleHDR()
   DX::Windowing()->SetAlteringWindow(false);
 
   // Re-create swapchain
-  if (hdrStatus != HDR_STATUS::HDR_TOGGLE_FAILED)
+  if (!isXbox && hdrStatus != HDR_STATUS::HDR_TOGGLE_FAILED)
   {
     CreateWindowSizeDependentResources();
 
     DX::Windowing()->NotifyAppFocusChange(true);
   }
 
+  // On Xbox set new color space in same swapchain
+  if (isXbox && hdrStatus != HDR_STATUS::HDR_TOGGLE_FAILED)
```

Comment:
Isn't CRendererBase::ProcessHDR supposed to take care of setting the swap chain color space? (it also sets the HDR10 metadata, which should also work on xbox now)

Reply:
`CRendererBase::ProcessHDR` only is used for video playback but HDR also needs activated from Kodi GUI (F11 key) because GUI is rendered with color space BT.2020 / tranfer PQ when HDR is On.

This code does the same in Windows x64:

https://github.com/xbmc/xbmc/blob/031dca9bd69d4e4fa2826690e23eea884b7e3a05/xbmc/rendering/dx/DeviceResources.cpp#L755-L756

The only difference is BT.709 color space is not need to specify because is the default when swap chain is created but could be replaced by:

```c++
if (m_IsHDROutput) 
   SetHdrColorSpace(DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020); 
else
   SetHdrColorSpace(DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709); 
```

on Xbox since is not recreated is need a call also for transition HDR --> SDR

Reply:
ok